File: java/src/test/java/problems201_300/ImplementQueueUsingStacksTest.java
package problems201_300;

import java.util.Objects;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertThat;

public class ImplementQueueUsingStacksTest extends SampledTest {
    @Test
    public void test() {
        for (ImplementQueueUsingStacksSample sample
            : getSamples(ImplementQueueUsingStacksSample[].class)) {
            ImplementQueueUsingStacks queue = null;
            for (int i = 0; i < sample.input.moveNames.length; i++) {
                int[] values = sample.input.moveValues[i];
                switch (sample.input.moveNames[i]) {
                    case "MyQueue":
                        queue = new ImplementQueueUsingStacks();
                        assertThat(sample.output[i])
                            .isNull();
                        break;
                    case "push":
                        Objects.requireNonNull(queue).push(values[0]);
                        assertThat(sample.output[i])
                            .isNull();
                        break;
                    case "peek":
                        assertThat(sample.output[i])
                            .isEqualTo(Objects.requireNonNull(queue).peek());
                        break;
                    case "pop":
                        assertThat(sample.output[i])
                            .isEqualTo(Objects.requireNonNull(queue).pop());
                        break;
                    case "empty":
                        assertThat(sample.output[i])
                            .isEqualTo(Objects.requireNonNull(queue).empty());
                        break;
                }
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/ImplementQueueUsingStacksTest.kt
package problems201_300

import com.google.common.truth.Truth.assertThat
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ImplementQueueUsingStacksTest : SampledTest() {
    @Test
    fun test() {
        getSamples<Array<ImplementQueueUsingStacksSample>>().forEach { (input, output, message) ->
            lateinit var queue: ImplementQueueUsingStacks
            for (i in input.moveNames.indices) {
                val values = input.moveValues[i]
                when (input.moveNames[i]) {
                    "MyQueue" -> {
                        queue = ImplementQueueUsingStacks()
                        assertThat(output[i])
                            .isNull()
                    }

                    "push" -> {
                        queue.push(values[0])
                        assertThat(output[i])
                            .isNull()
                    }

                    "peek" ->
                        assertThat(output[i])
                            .isEqualTo(queue.peek())

                    "pop" ->
                        assertThat(output[i])
                            .isEqualTo(queue.pop())

                    "empty" ->
                        assertThat(output[i])
                            .isEqualTo(queue.empty())
                }
            }
        }
    }
}
